sources/src/main/java/com/google/solutions/jitaccess/core/services/RoleDiscoveryService.java

```diff
@@ -108,44 +114,52 @@ public Options getOptions() {
-    // applies.
-    //
-    // The risk here is that the list of resources grows so large that we're hitting
-    // the limits of the API, in which case it starts truncating results. To
-    // mitigate this risk, filter on a permission that:
-    //
-    // - only applies to projects, and has no meaning on descendent resources
-    // - represents the lowest level of access to a project.
-    //
-    var analysisResult = this.assetInventoryAdapter.findAccessibleResourcesByUser(
-      this.options.scope,
-      user,
-      Optional.of("resourcemanager.projects.get"),
-      Optional.empty(),
-      true);
-
-    //
-    // Consider permanent and eligible bindings.
-    //
-    var roleBindings = findRoleBindings(
-      analysisResult,
-      condition -> condition == null ||
-        JitConstraints.isJitAccessConstraint(condition) ||
-        JitConstraints.isMultiPartyApprovalConstraint(condition),
-      evalResult -> evalResult == null ||
-        "TRUE".equalsIgnoreCase(evalResult) ||
-        "CONDITIONAL".equalsIgnoreCase(evalResult));
-
-    return roleBindings
-      .stream()
-      .map(b -> ProjectId.fromFullResourceName(b.fullResourceName))
-      .collect(Collectors.toSet());
+    if(this.options.listAllAvailableProjectsIn == null) {
+      //
+      // NB. To reliably find projects, we have to let the Asset API consider
+      // inherited role bindings by using the "expand resources" flag. This
+      // flag causes the API to return *all* resources for which an IAM binding
+      // applies.
+      //
+      // The risk here is that the list of resources grows so large that we're hitting
+      // the limits of the API, in which case it starts truncating results. To
+      // mitigate this risk, filter on a permission that:
+      //
+      // - only applies to projects, and has no meaning on descendent resources
+      // - represents the lowest level of access to a project.
+      //
+      var analysisResult = this.assetInventoryAdapter.findAccessibleResourcesByUser(
+        this.options.scope,
+        user,
+        Optional.of("resourcemanager.projects.get"),
+        Optional.empty(),
+        true);
+
+      //
+      // Consider permanent and eligible bindings.
+      //
+      var roleBindings = findRoleBindings(
+        analysisResult,
+        condition -> condition == null ||
+          JitConstraints.isJitAccessConstraint(condition) ||
+          JitConstraints.isMultiPartyApprovalConstraint(condition),
+        evalResult -> evalResult == null ||
+          "TRUE".equalsIgnoreCase(evalResult) ||
+          "CONDITIONAL".equalsIgnoreCase(evalResult));
+
+      return roleBindings
+        .stream()
+        .map(b -> ProjectId.fromFullResourceName(b.fullResourceName))
+        .collect(Collectors.toSet());
+      }
+    else {
+      // Used as fallback if listAllAvailableProjects is set to true and Asset API is not working fast enough.
```

Comment:
nit: Not sure if we should call it "fallback" if it's a configuration option.

Reply:
Fixed